core/src/mock_alpenglow_consensus.rs

```diff
@@ -0,0 +1,754 @@
+#![allow(warnings)]
+use {
+    crate::consensus::Stake,
+    bytemuck::{Pod, Zeroable},
+    chrono::TimeDelta,
+    crossbeam_channel::{bounded, Receiver, Sender},
+    dashmap::DashMap,
+    futures::future::err,
+    serde::de::DeserializeOwned,
+    serde_bytes::Deserialize,
+    solana_clock::Slot,
+    solana_gossip::{
+        cluster_info::ClusterInfo,
+        epoch_specs::{self, EpochSpecs},
+    },
+    solana_keypair::Keypair,
+    solana_packet::{Meta, Packet, PACKET_DATA_SIZE},
+    solana_pubkey::{Pubkey, PUBKEY_BYTES},
+    solana_runtime::bank::Bank,
+    solana_signature::{Signature, SIGNATURE_BYTES},
+    solana_signer::Signer,
+    solana_streamer::{recvmmsg::recv_mmsg, sendmmsg::batch_send},
+    solana_turbine::cluster_nodes,
+    static_assertions::const_assert,
+    std::{
+        collections::HashMap,
+        io::Error,
+        net::{SocketAddr, SocketAddrV4, UdpSocket},
+        sync::{
+            atomic::{AtomicBool, AtomicU16, Ordering},
+            Arc, Mutex,
+        },
+        thread::{self, JoinHandle},
+        time::{Duration, Instant},
+    },
+};
+
+/// This is a placeholder that is only used for load-testing.
+/// This is not representative of the actual alpenglow implementation.
+pub(crate) struct MockAlpenglowConsensus {
+    sender_thread: JoinHandle<()>,
+    listener_thread: JoinHandle<()>,
+    state: Arc<[Mutex<SharedState>; 3]>, // internal state of the test
+    highest_slot: Slot,                  // highest slot we have observed so far
+    safety_latch_engaged: bool, // will be set to true if root bank is too far behind the slot we are voting for
+    should_exit: Arc<AtomicBool>,
+    verify_signatures: Arc<AtomicBool>,
+    packet_size: Arc<AtomicU16>,
+    // external state
+    epoch_specs: EpochSpecs,
+    cluster_info: Arc<ClusterInfo>,
+    // control of internal threadpool that handles test timings
+    slot_sender: Option<Sender<Slot>>,
+    thread_pool: [JoinHandle<()>; 3],
```

Comment:
nit: we could put a big warning here this is a separate thread pool so that our tests will never grab too much CPU from the main tasks. Just in case we forget about it and someone wants to fold this into the main thread pool.

I thought you would use `ThreadPool`, but this also works.